DatadogCore/Sources/Datadog.swift

```diff
@@ -345,6 +345,9 @@ public struct Datadog {
             ?? configuration.bundle.object(forInfoDictionaryKey: "CFBundleVersion") as? String
             ?? "0.0.0"
 
+        let applicationBuildNumber = configuration.bundle.object(forInfoDictionaryKey: "CFBundleVersion") as? String
+            ?? "0"
```

Comment:
❓ 
I don't recall exactly (it has been long time since I published an app) but can this be different than App Store build number?

If yes, then I expect build number (as well as application version) part of configuration (plain strings). 

So the final resolutions looks like
1. try from configuration string value
2. try from bundle
3. null (either missing from the payload or explicitly marked as `null` json value.)

❓ this is this a required field?

Reply:
The `build_version` is a required field, same for the `version` field. We have not exposed an application version parameter in the core config, and I believe it should be the same for the build number.

They are both corresponding to values in the app store, but since we are getting them from the `Bundle` with optional casting to `String`, we have a fallback value that won't be reached in production.

Reply:
> I don't recall exactly (it has been long time since I published an app) but can this be different than App Store build number?

I explained our rationale [here](https://github.com/DataDog/dd-sdk-ios/issues/1026#issuecomment-1288739533). In a nutshell:
* `CFBundleShortVersionString` = [`MARKETING_VERSION`](https://developer.apple.com/documentation/xcode/build-settings-reference#Marketing-Version) = what users see in App Store
* `CFBundleVersion` = [`CURRENT_PROJECT_VERSION`](https://developer.apple.com/documentation/xcode/build-settings-reference#Current-Project-Version) = what devs see in App Store